vNext/src/Orleans/Shims/AppDomain.cs

```diff
@@ -31,19 +34,31 @@ private AppDomain()
             //};
         }
 
-        public Assembly[] GetAssemblies()
+        private Lazy<Assembly[]> assembliesList = new Lazy<Assembly[]>(() =>
         {
-            // TODO: replace with IAssemblyCatalog or something like that.
-            Assembly[] assemblies =
+            string path = ".";
+            var assemblyFiles = Directory.EnumerateFiles(path, "*.dll");
```

Comment:
Heh, actually I had that originally (plus unescaping the uri string) but decided to simplify it like this for now, since it's only temporary. I wouldnt mind changing to it again if that's preferable though